Source/scrollrt.cpp

```diff
@@ -852,10 +852,9 @@ void DrawDungeon(const Surface &out, Point tilePosition, Point targetBufferPosit
 			int px = targetBufferPosition.x - CalculateWidth2(pDeadGuy->width);
 			const byte *pCelBuff = pDeadGuy->data[(bDead >> 5) & 7];
 			assert(pCelBuff != nullptr);
-			const auto *frameTable = reinterpret_cast<const uint32_t *>(pCelBuff);
-			int frames = SDL_SwapLE32(frameTable[0]);
+			const uint32_t frames = LoadLE32(pCelBuff);
 			int nCel = pDeadGuy->frame;
-			if (nCel < 1 || frames > 50 || nCel > frames) {
+			if (nCel < 1 || frames > 50 || nCel > static_cast<int>(frames)) {

```

Comment:
Any reason we can't make frame/nCel unsigned?

Reply:
I'm not sure, we probably could but best leave this for another PR